neps/nep-0568.md

```diff
@@ -112,6 +112,20 @@ Cold storage uses the same mapping strategy to manage shard state during reshard
 
 This approach minimizes complexity while maintaining consistency across hot and cold storage.
 
+#### State cleanup
+
+Since [Stateless Validation][NEP-508], all shards tracking is no longer required. Currently, shard cleanup has not been implemented. We propose a cleanup mechanism that will also handle post-resharding cleanup.
+
+When garbage collection removes the last block of an epoch from the canonical chain, we determine which shards were tracked during that epoch by examining the shards present in `TrieChanges` at that block. Similarly, we collect  information on shards tracked in subsequent epochs, up to the present one. A shard State is removed only if:
```

Comment:
nit: two spaces between "collect" and "information"

Reply:
fixed in https://github.com/near/NEPs/commit/333544c7499f10b61773c81779f4c8116984ebd5